Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -668,4 +674,32 @@ def test_search
   ensure
     desc_cache.unlink
   end
+
+  def test_bundle
+    skip_unless_test_cmd_taps
+    HOMEBREW_REPOSITORY.cd do
+      shutup do
+        system "git", "init"
+        system "git", "commit", "--allow-empty", "-m", "This is a test commit"
+      end
+    end
+
+    mktmpdir do
+      FileUtils.touch "Brewfile"
+      assert_equal "The Brewfile's dependencies are satisfied.",
+        cmd("bundle", "check")
+    end
+  ensure
+    (HOMEBREW_REPOSITORY/".git").rmtree
```

Comment:
Probably not for this PR, but we could also consider adding a method that basically does “register this path for cleanup” and that is called every time a new path in a test is used. This would delegate the removal of files and directories to `teardown` instead of relying on `ensure` in every single test method (where this file/directory cleanup is dominating). I'm not sure though, whether this is the right approach or if it makes the tests less readable.